zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -1038,8 +1038,10 @@ private boolean inlineRemove(String id, boolean initiator) throws IOException {
       removed = true;
     }
 
-    File localRepoDir = new File(conf.getInterpreterLocalRepoPath() + "/" + id);
-    FileUtils.deleteDirectory(localRepoDir);
+    File localRepoPath = new File(conf.getInterpreterLocalRepoPath());
```

Comment:
It is not necessary to compare interpreter setting id again here, just move this to if loop should fix it.

Reply:
```
if (interpreterSettings.containsKey(id)) {
      InterpreterSetting intp = interpreterSettings.get(id);
      intp.close();
      removeInterpreterSetting(id);
      if (initiator) {
        // Event initiator saves the file
        // Cluster event accepting nodes do not need to save files repeatedly
        saveToFile();
      }
      FileUtils.deleteDirectory(new File(localRepoPath, id));
      removed = true;
    }

```

Do you mean to do it this way?

Reply:
@zjffdu

Reply:
That's right